src/loaders/genome_collection/parse_PMI_biolog_data.py

```diff
@@ -0,0 +1,233 @@
+
+GROWTH_MEDIA_COL_NAME = 'growth_media'
+STRAIN_DESIGNATION_COL_NAME = 'strain designation'
+DEFAULT_MEDIA = 'MOPS minimal media'
+
+
+def _read_excel_as_df(excel_file: Path, sheet_name=0) -> pd.DataFrame:
+    # Read the Excel file into a DataFrame
+
+    try:
+        df = pd.read_excel(excel_file, sheet_name=sheet_name)
+
+        return df
+
+    except FileNotFoundError:
+        print(f"File '{excel_file}' not found.")
+
+
+def _find_matching_media(column_name, media_mapping):
+    # find the matching key from media_mapping
+    for key in media_mapping.keys():
+        if column_name.endswith(key):
+            return media_mapping[key]
+    return DEFAULT_MEDIA
+
+
+def _read_biolog_data(biolog_data_file: Path) -> pd.DataFrame:
+    # Read the Biolog data file into a DataFrame and do some data cleaning
+
+    data_df = _read_excel_as_df(biolog_data_file)
+
+    # NOTE: below logic is specific to the Biolog data file (PMI strain BiologSummary.xlsx)
+    # link to the file: https://docs.google.com/spreadsheets/d/1QmC6UHWOEVfpmrveRBl_izictbNeN1PA/edit#gid=1135979967
+
+    # Drop unused columns
+    data_df = data_df.drop(columns=['# pos. strains'])
+
+    # Make the first row ('strain designation') the column names
+    data_df.columns = data_df.loc[1]
+
+    # Iterate through the "strain designation" column and parse the growth media description
+    media_mapping = {}
+    for value in data_df[STRAIN_DESIGNATION_COL_NAME]:
+        if isinstance(value, str):
+            matches = re.findall(r'\*+', value)  # find '*' and '**' patterns
+            if matches:
+                num_stars = max(len(match) for match in matches)
+                media_description = value.replace('*', '').strip()
+                media_mapping['*' * num_stars] = media_description
+
+    # Reorder media_mapping by the number of '*' characters in reverse order.
+    # This step is necessary to ensure that _find_matching_media can match the longest key
+    # because the keys ('*' and '**') are inclusive.
+    media_mapping = dict(sorted(media_mapping.items(), key=lambda x: len(x[0]), reverse=True))
+
+    # Set the growth media row with the matching media description
+    data_df.loc[GROWTH_MEDIA_COL_NAME] = data_df.columns.map(lambda x: _find_matching_media(x, media_mapping))
+
+    # Drop unused rows ('total cpds. Utilized', 'strain designation' and unused rows at the end.)
+    # NOTE: Drop 'strain designation' here because it also becomes the column names at the previous step.
+    #       Row 'Carbon Source' is used as the index upon reading the file.
+    #       Therefore, dataframe row index 0 starts at 'total cpds. Utilized' and
+    #       the last row index with data is 193 (row 195 '3-Hydroxy 2-Butanone' in the Excel sheet).
+    data_df = data_df.drop(index=[0, 1, 194, 195, 196, 197, 198])
+
+    # Make the first column ('strain designation') the index
+    data_df.loc[GROWTH_MEDIA_COL_NAME, STRAIN_DESIGNATION_COL_NAME] = GROWTH_MEDIA_COL_NAME
```

Comment:
I'm not understanding what's going on here

Reply:
added a comment to explain this. It was needed because I want to set `strain designation` as index in the next step. And currently, the cell was filled with the default media description, but we want the GROWTH_MEDIA_COL_NAME as column name.

Reply:
Maybe something like
```
Switch the value of the cell in row `growth_media` and column `strain designation` from the default media string to `growth_media`. This is needed so that when we make `strain designation` the index in the next step `growth_media` will be a row in that index.
```

I think I've got that right but feel free to improve

Reply:
👍

Reply:
:+1: